builder/openstack/step_create_volume.go

```diff
@@ -111,24 +109,9 @@ func (s *StepCreateVolume) Cleanup(state multistep.StateBag) {
 		return
 	}
 
-	// Wait for volume to become available.
-	status, err := GetVolumeStatus(blockStorageClient, s.volumeID)
-	if err != nil {
-		ui.Error(fmt.Sprintf(
-			"Error getting the volume information. Please delete the volume manually: %s", s.volumeID))
-		return
-	}
-
-	if status != "available" {
-		ui.Say(fmt.Sprintf(
-			"Waiting for volume %s (volume id: %s) to become available...", s.VolumeName, s.volumeID))
-		if err := WaitForVolume(blockStorageClient, s.volumeID); err != nil {
-			ui.Error(fmt.Sprintf(
-				"Error getting the volume information. Please delete the volume manually: %s", s.volumeID))
-			return
-		}
-	}
 	ui.Say(fmt.Sprintf("Deleting volume: %s ...", s.volumeID))
+
+	// Delete the volume in any status if exists.
 	err = volumes.Delete(blockStorageClient, s.volumeID, volumes.DeleteOpts{}).ExtractErr()
 	if err != nil {
 		ui.Error(fmt.Sprintf(
```

Comment:
Just to be more explicit while we're working on this code, I'd suggest adding the error to the error message.

```suggestion
		ui.Error(fmt.Sprintf("Error cleaning up volume %q: %s. This may need manual deletion.", s.volumeID, err))
```